### ISSUE TITLE
TypeError: TarFile.extractall() got an unexpected keyword argument 'filter' when running openai migrate

### ISSUE THREAD

**Issue by imene-swaan** [openai/openai-python#1435]
### Confirm this is an issue with the Python library and not an underlying OpenAI API

- [X] This is an issue with the Python library

### Describe the bug

When running the openai migrate command to update my code to the latest version due to deprecated functions, I encountered a TypeError. The error message indicates that the TarFile.extractall() method received an unexpected keyword argument filter. The error seems to originate from the migrate.py file within the openai package, specifically from the line attempting to call archive.extractall with an unsupported filter argument.

### To Reproduce

1. Run this command  in your terminal:
`openai migrate`
2. Observe the error message:

`Retrieving Grit CLI metadata from https://api.keygen.sh/v1/accounts/custodian-dev/artifacts/marzano-macos-arm64
Traceback (most recent call last):
  File "path/to/openai", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "path/to/_cli.py", line 129, in main
    _main()
  File "path/to/_cli.py", line 209, in _main
    parsed.func(
  File "path/to/migrate.py", line 53, in migrate
    grit_path = install()
                ^^^^^^^^^
  File "path/to/migrate.py", line 141, in install
    archive.extractall(unpacked_dir, filter="data")
TypeError: TarFile.extractall() got an unexpected keyword argument 'filter'
`

### OS

macos

### Python version

3.11.3

### Library version

1.30.1

**Comment by rattrayalex** [comment]
cc @morgante can you take a look?

**Comment by morgante** [comment]
It looks like this [this script](https://github.com/openai/openai-python/blob/7fff7c842918e718e1e9dc3954f0e66aebfc2601/src/openai/cli/_tools/migrate.py#L141) is relying on `filter`, which was only added in Python 1.12.

**Comment by tobwen** [comment]
> It looks like this [this script](https://github.com/openai/openai-python/blob/7fff7c842918e718e1e9dc3954f0e66aebfc2601/src/openai/cli/_tools/migrate.py#L141) is relying on `filter`, which was only added in Python 1.12.

For the records, 3.12, hehe https://docs.python.org/3/library/tarfile.html#tarfile-extraction-filter

**Comment by Yooolin** [comment]
I have also encountered this error.  when can it probably be fixed?

**Comment by lostsync** [comment]
this worked for me

`python $(which openai) migrate`

if you're using conda or venv, openai package may be using a different python version than you are at cli. running python explicitly will work around it.

**Comment by outpark** [comment]
Any fix yet?

**Comment by thunderstudying** [comment]
> > It looks like this [this script](https://github.com/openai/openai-python/blob/7fff7c842918e718e1e9dc3954f0e66aebfc2601/src/openai/cli/_tools/migrate.py#L141) is relying on `filter`, which was only added in Python 1.12.
> 
> For the records, 3.12, hehe https://docs.python.org/3/library/tarfile.html#tarfile-extraction-filter
> 
> But I've got the same issue.

According to the quote reply,

update your python to `3.12.x` through conda or pip
```
conda update python
pip install --upgrade python
```
then execute the command again
```
openai migrate
```

**Comment by tobwen** [comment]
> update your python to `3.12.x` through conda or pip

Sound like a packaging issue to me. Shouldn't `pip` inform that this version of `openai-python` is too new for my version of Python (or vice-versa)?

**Comment by RobertCraigie** [comment]
Really sorry for the delay here, this will be fixed in the next release (`v1.35.5`) https://github.com/openai/openai-python/pull/1501.
